tensorflow/contrib/tensorrt/convert/convert_nodes.cc

```diff
@@ -910,6 +943,12 @@ tensorflow::Status BinaryTensorOpWeight(
   auto dims_w = weights.shape_;
   auto dims_t = tensor->getDimensions();
 
+  // TODO(jie): addScale checks for input tensor dimension
+  if (dims_t.nbDims != 3) {
```

Comment:
Can we put all checks like this into a method and call it in the segmenter, so we can avoid putting nodes into the trt engine at the very beginning, instead of delaying the check until here?

Reply:
We have undergoing refactoring work on the segmenter. It is to improve the robustness of the converter and expand checking only the op types to consider attributes as well as input types.

So that will come in as future PRs.